src/core/core.controller.js

```diff
@@ -627,8 +627,13 @@ module.exports = function(Chart) {
 			}
 
 			// On Hover hook
-			if (hoverOptions.onHover) {
-				hoverOptions.onHover.call(me, me.active);
+			if (e.type === 'mousemove') {
+			    if (hoverOptions.onHover) {
+			        hoverOptions.onHover.call(me, me.active);
+			    }
+			    if (me.legend && me.legend.handleEvent && me.legend.options.onHover) {
```

Comment:
I'm even thinking that this `eventHandler` should simply forward the event to the legend `handleEvent` and so delegate all the event logic to the legend. Meaning, no `e.type` check here, simply:

    if (me.legend && me.legend.handleEvent) {
        me.legend.handleEvent(e);
    }

Which also mean that the legend `handleEvent` should discard useless events asap:

    handleEvent: function(e) {
        var me = this;
        var type = e.type;
        // ...
   
        type = type === 'mouseup'? 'click : type;
        if (e.type !== 'click' && type !== 'mousemove') {
            return;
        }

        // ...
    }